sdk/ml/azure-ai-ml/azure/ai/ml/operations/_component_operations.py

```diff
@@ -640,13 +679,20 @@ def _resolve_dependencies_for_pipeline_component_jobs(
             component,
             extra_operations=[
                 self._set_default_display_name_for_anonymous_component_in_node,
+                partial(self._try_resolve_node_level_task_for_parallel_node, resolver=resolver),
             ]
         )
 
         # cache anonymous component only for now
         # request level in-memory cache can be a better solution for other type of assets as they are
         # relatively simple and of small number of distinct instances
-        component_cache = CachedNodeResolver(resolver=resolver)
+        component_cache = CachedNodeResolver(
+            resolver=resolver,
+            subscription_id=self._subscription_id,
+            resource_group_name=self._resource_group_name,
+            workspace_name=self._workspace_name,
+            registry_name=self._registry_name,
+        )
```

Comment:
Have a question for L709, will environment be resolved to arm id before checking cache?

Reply:
No. As explained [here](https://github.com/Azure/azure-sdk-for-python/pull/28072/files/1434e8ff747b6e6d89cccf7d0d2d9f677e5da8fb#r1058709107), it should be thread-safe.

Reply:
But such dependencies resolution will impact the efficiency of cache hit across layers, so not sure whether we should guarantee that all other dependencies are resolved in extra operations

Reply:
Interesting... Resolution of environment will be skipped in some cases in [early recording](https://github.com/Azure/azure-sdk-for-python/blob/ca26a259193d985b65e9e181a554e1529ccf5751/sdk/ml/azure-ai-ml/tests/recordings/dsl/e2etests/test_dsl_pipeline.pyTestDSLPipelinetest_distribution_components.json#L343) (as early as 11.14)

Reply:
Got the reason. updated in latest commit